### DISCUSSION TITLE
How to allow access to photo library on iOS

### DISCUSSION THREAD

**Discussion by csondi10** [leancodepl/patrol#1944] [Q&A]
Hey guys,

I'd need help with a native permission dialog. How to tap on "Allow full access"? 
Further question would be then, how to select an image? 😄 

(iOS 17.0 simulator)

![gallery](https://github.com/leancodepl/patrol/assets/105285564/3723a93c-2621-42f5-a125-33e320fe6546)

**Comment by bartekpacia** [comment]
Try this:

```dart
import 'dart:io' as io;

// ...

await $.native.tap(
  Selector(text: 'Allow full access'),
  appId: io.Platform.isIOS ? 'com.apple.mobileslideshow' : null,
);
```

IDs of apps installed on iOS simulator can be obtained with `xcrun simctl listapps booted`.

**Reply by csondi10** [reply]
Thank you, I'll test it. Is it possible to include this **xcrun simctl listapps booted** information in the docs?

**Reply by csondi10** [reply]
Hi @bartekpacia ,

I tried your suggestion but unfortunately had no luck. It basically did nothing.
Can you provide further suggestions, please?

**Reply by bartekpacia** [reply]
Hmm, then this is likely a bug. Please create a new bug report issue and fill out the template.

**Reply by bartekpacia** [reply]
For future readers: see https://github.com/leancodepl/patrol/issues/1994#issuecomment-1855836290
